### PR TITLE
AESinkAudioTrack: Search correct buffer by incrementing in min_bufferunits

### PR DESCRIPTION
**Pull request by fritsch**
This is a further refinement, to land nearer to 200 than 300 by incrementing min_buffer

### CODE REVIEW

**Comment by fritsch** [comment]
Don't merge - good time to revisit all the hacks from the past. I will use this PR to rework Audiotrack.

**Comment by fritsch** [comment]
Tested both methods: IEC and RAW on Firetv 4K.

Could not test: 
DTS-HD-MA (IEC + RAW)
TrueHD (IEC + RAW)

Test works

File: xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp [hunk 1]
```diff
@@ -594,45 +590,25 @@ void CAESinkAUDIOTRACK::GetDelay(AEDelayStatus& status)
...
-    if (m_extTimer.MillisLeft() > 0)
-    {
-      const double d = GetMovingAverageDelay(GetCacheTotal());
-      status.SetDelay(d);
-      return;
-    }
-  }
-
-  double gone = (double) normHead_pos / (double) m_sink_sampleRate;
+  double gone = (double)m_headPos / (double)m_sink_sampleRate;
```
Comment (Rechi) [inline xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp]: Can you replace the C style casts with C++ style casts as you already touched the line?

File: xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp [hunk 2]
```diff
@@ -738,29 +709,50 @@ unsigned int CAESinkAUDIOTRACK::AddPackets(uint8_t **data, unsigned int frames,
       loop_written = 0;
     }
   }
-  unsigned int written_frames = (unsigned int) (written/m_format.m_frameSize);
+  unsigned int written_frames = (unsigned int)(written / m_format.m_frameSize);
```
Comment (Rechi) [inline xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp]: Can you replace the C style cast with a C++ style cast as you already touched the line?

File: xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp (hunk 1 above)
Comment (fritsch) [reply xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp]: Yes - will do. Thanks.

File: xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp (hunk 2 above)
Comment (fritsch) [reply xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp]: Yes - will do. Thanks.

**Comment by fritsch** [comment]
jenkins build this please

**Comment by fritsch** [comment]
jenkins build this please

**Comment by fritsch** [comment]
UWP build failure not related.

@HitcherUK tested it. Will wait for @da-anda who is testing as well and then deliver it.

**Comment by fritsch** [comment]
jenkins build this please

**Comment by fritsch** [comment]
Win64 UWP not related.

**Pull request merged**